src/pkg/path/path.go

```diff
@@ -317,6 +319,21 @@ func Split(segment string) []string {
 	return res
 }
 
+func ArePathsEquivalent(path1, path2 string) bool {
+	normalizedPath1 := strings.TrimSpace(filepath.Clean(path1))
+	normalizedPath2 := strings.TrimSpace(filepath.Clean(path2))
+
+	normalizedPath1 = strings.TrimSuffix(normalizedPath1, string(filepath.Separator))
+	normalizedPath2 = strings.TrimSuffix(normalizedPath2, string(filepath.Separator))
+
+	return normalizedPath1 == normalizedPath2
+}
+
+func IsValidPath(path string) bool {
```

Comment:
The name is a bit misleading, since many inputs to this func could hold valid paths without passing the stat check.  This name makes it a bit more clear what we're checking.
```suggestion
func DoesPathExistInOS(path string) bool {
```